homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -0,0 +1,234 @@
+"""
+Sensor to collect the reference daily prices of electricity ('PVPC') in Spain.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/integrations/pvpc_hourly_pricing/
+"""
+from datetime import timedelta
+import logging
+from random import randint
+from typing import Optional
+
+from aiopvpc import PVPCData
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import ENTITY_ID_FORMAT, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import (
+    async_track_point_in_time,
+    async_track_time_change,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+import homeassistant.util.dt as dt_util
+
+from . import SENSOR_SCHEMA
+from .const import ATTR_TARIFF, DEFAULT_TARIFF, DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(SENSOR_SCHEMA.schema)
+
+ATTR_PRICE = "price"
+ICON = "mdi:currency-eur"
+UNIT = "€/kWh"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_devices, discovery_info=None
+):
+    """
+    Set up the electricity price sensor as a sensor platform.
+
+    ```yaml
+    sensor:
+      - platform: pvpc_hourly_pricing
+        name: pvpc_manual_sensor
+        tariff: normal
+
+      - platform: pvpc_hourly_pricing
+        name: pvpc_manual_sensor_2
+        tariff: discrimination
+        timeout: 8
+    ```
+    """
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, data=config, context={"source": config_entries.SOURCE_IMPORT}
+        )
+    )
+    return True
+
+
+async def update_listener(hass: HomeAssistant, entry: config_entries.ConfigEntry):
+    """Update selected tariff for sensor."""
+    hass.async_create_task(hass.config_entries.async_reload(entry.entry_id))
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry, async_add_entities
+):
+    """Set up the electricity price sensor from config_entry."""
+    if not config_entry.update_listeners:
+        config_entry.add_update_listener(update_listener)
+
+    name = config_entry.data[CONF_NAME]
+    entity_id = async_generate_entity_id(ENTITY_ID_FORMAT, name, hass=hass)
```

Comment:
Please don't pass the entity_id in as this can be changed in the UI

<img width="587" alt="Screen Shot 2020-03-19 at 9 56 38 AM" src="https://user-images.githubusercontent.com/663432/77080998-01098c00-69c8-11ea-87be-cdcb317308b0.png">

Reply:
Done, now passing the unique_id (made from tariff)